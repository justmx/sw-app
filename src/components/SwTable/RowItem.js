import React from "react";

const RowItem = ({ person, onClick }) => {
  if (!person) return null;
  const { name, height, mass } = person;

  return (
    <div className="row" onClick={onClick}>
      <div className="cell">{name}</div>
      <div className="cell">{height}</div>
      <div className="cell">{mass}</div>
    </div>
  );
};

export default RowItem;
