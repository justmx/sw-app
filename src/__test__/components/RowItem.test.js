import React from "react";
import { shallow } from "enzyme";
import RowItem from "../../components/SwTable/RowItem";

const clickFn = jest.fn();
const person = { name: "Jack", height: "177", mass: 168 };

describe("RowItem", () => {
  it("should render row item", () => {
    const wrapper = shallow(<RowItem person={person} onClick={clickFn} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should return null if no props", () => {
    const wrapper = shallow(<RowItem />);
    expect(wrapper.type()).toEqual(null);
  });

  it("should render div with name Jack", () => {
    const wrapper = shallow(<RowItem person={person} onClick={clickFn} />);
    const nameDiv = <div className="cell">Jack</div>;
    expect(wrapper.contains(nameDiv)).toEqual(true);
  });

  it("should render 3 cell div", () => {
    const wrapper = shallow(<RowItem person={person} onClick={clickFn} />);
    expect(wrapper.find("div.cell")).toHaveLength(3);
  });

  it("should fire a function when click", () => {
    const wrapper = shallow(<RowItem person={person} onClick={clickFn} />);
    wrapper.find(".row").simulate("click");
    expect(clickFn).toHaveBeenCalled();
  });
});
