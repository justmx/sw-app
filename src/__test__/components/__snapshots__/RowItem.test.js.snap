// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowItem should render row item 1`] = `
<div
  className="row"
  onClick={[MockFunction]}
>
  <div
    className="cell"
  >
    Jack
  </div>
  <div
    className="cell"
  >
    177
  </div>
  <div
    className="cell"
  >
    168
  </div>
</div>
`;
